tools/patch-modify

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import sys
+
+def loadPatch(filename):
+    """ Load the patch from file
+    """
+    text = ""
+    with open(filename, "r") as fp:
+        text = fp.read()
+
+    # make sure this looks like a patch file
+    if not text.startswith("From "):
+        return None
+
+    return text
+
+def savePatch(filename, patch):
+    """ Save the patch to file
+    """
+    with open(filename, "w") as fp:
+        fp.write(patch)
+
+def splitPatch(text):
+    """ Split the patch into header, patch parts and version
+    """
+    # cut off the version
+    versionPos = text.rfind("\n--")
+    version = text[versionPos:]
+    text = text[:versionPos]
```

Comment:
`versionPos` might be -1 if there is no signature, then this would cripple both version and text. You can use something like
```python
try:
    (text, version) = text.rsplit('\n-- ', 1)
except ValueError:
    version = ''
```